code/modules/reagents/chems/chems_medicines.dm

```diff
@@ -359,3 +359,38 @@
 /decl/material/liquid/oxy_meds/affect_blood(var/mob/living/M, var/removed, var/datum/reagents/holder)
 	M.add_chemical_effect(CE_OXYGENATED, 1)
 	holder.remove_reagent(/decl/material/gas/carbon_monoxide, 2 * removed)
+
+/decl/material/liquid/clotting_agent
+	name = "clotting agent"
+	uid = "chem_clotting_agent"
+	lore_text = "A medication used to rapidly clot internal hemorrhages by increasing the effectiveness of platelets."
+	metabolism = REM * 0.5
+	overdose = REAGENTS_OVERDOSE * 0.5
+	color = "#4246c7"
+	scannable = TRUE
+
+/decl/material/liquid/clotting_agent/affect_blood(mob/living/M, removed, datum/reagents/holder)
+	SET_STATUS_MAX(M, STAT_BLURRY, 30)
+	M.add_chemical_effect(CE_BLOCKAGE, (15 + REAGENT_VOLUME(holder, type))/100)
+	for(var/obj/item/organ/external/E in M.get_external_organs())
```

Comment:
Maybe only check `bad_external_organs` so it doesn't loop over all organs for victim?